### PR TITLE
[1.16] Some fixes for script runs and basic `/ct examples` command 

### PR DESCRIPTION
**Pull request by kindlich**
Do not merge until https://github.com/ZenCodeLang/ZenCode/pull/2 is merged, since this PR relies on the new constructor on ParseException

### CODE REVIEW

File: src/main/java/com/blamejared/crafttweaker/api/zencode/impl/util/PositionUtil.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     * <p>
+     * The Position's file will always be a {@link VirtualSourceFile} so you cannot access the script content!
+     *
+     * @return The position, or {@link CodePosition#UNKNOWN}
+     */
+    public static CodePosition getZCScriptPositionFromStackTrace() {
+        final StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
+        final StackTraceElement stackTraceElement = Arrays.stream(stackTrace)
+                .filter(element -> element.getFileName() != null)
+                .filter(element -> element.getFileName().endsWith(".zs"))
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/crafttweaker/api/zencode/impl/util/PositionUtil.java:22]: How does this handle calling another script from a script?

File: src/main/resources/data/crafttweaker/scripts/crafttweaker_tags.zs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+<tag:items:crafttweaker:all_pickaxes>.add(<item:minecraft:stone_pickaxe>);
+<tag:items:crafttweaker:all_pickaxes>.add(<item:minecraft:iron_pickaxe>);
+<tag:items:crafttweaker:all_pickaxes>.add(<item:minecraft:golden_pickaxe>);
+<tag:items:crafttweaker:all_pickaxes>.add(<item:minecraft:diamond_pickaxe>);
+<tag:items:crafttweaker:all_pickaxes>.add(<item:minecraft:netherite_pickaxe>);
+
+var pickaxeAll = <tag:items:crafttweaker:all_pickaxes>;
+//Then you can do something like 1 Cobblestone Stairs turns into 3 Cobblestone, but it needs a pickaxe, and it consumed 2 durability.
+
+craftingTable.addShapeless("stairs to cobble", <item:minecraft:cobblestone>*3, [<item:minecraft:cobblestone_stairs>, pickaxeAll.asIIngredient().transformDamage(2)]);
```
Comment (pupnewfster) [inline src/main/resources/data/crafttweaker/scripts/crafttweaker_tags.zs]: If this is meant to be an example, shouldn't the recipe's name not have spaces in it so it doesn't have a warning printed and convert them to underscores and instead is correct from the start?

Comment (kindlich) [reply src/main/resources/data/crafttweaker/scripts/crafttweaker_tags.zs]: Good catch!

File: src/main/java/com/blamejared/crafttweaker/api/zencode/impl/util/PositionUtil.java (hunk 1 above)
Comment (kindlich) [reply src/main/java/com/blamejared/crafttweaker/api/zencode/impl/util/PositionUtil.java:22]: It prints the topmost stack. So say we have
helperScript.zs
```zs
import crafttweaker.api.item.IItemStack;
import crafttweaker.api.item.IIngredient;

public function createPackingRecipe(input as IIngredient, output as IItemStack) as void {
	craftingTable.addShaped("packagingRecipe_" + output.registryName.path, output, [[input, input, input], [input, input, input], [input, input, input]]);
}
```

and callerScript.zs
```zs
createPackingRecipe(<item:minecraft:iron_block>, <item:minecraft:bedrock>);
```

So callerScript -> helperScript -> IAction would mean that `helperScript.zs:5` would be returned.

**Pull request merged**